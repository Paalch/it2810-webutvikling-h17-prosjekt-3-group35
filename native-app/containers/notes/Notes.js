import React from 'react';
import StyleSheet from 'react-native';
import Text from 'react-native';
import View from 'react-native';

class Notes extends React.Component {
  render () {
    return (
      <View Style={styles.notesContainer}>
        <link  href="http://fonts.googleapis.com/css?family=Reenie+Beanie:regular"
               rel="stylesheet"
               type="text/css"/>
        <View Style={styles.notesBoxHeader}>
          <Text>Notes</Text>
        </View>
        <View Style={styles.notesBoxBody}>
          <Text>Lorem ipsum</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  notesContainer: {
    backgroundColor: "#FFFFA5",
    padding: 30,
    shadowColor: 'rgba(33,33,33,.7)',
      shadowOpacity: 7,
      shadowOffset: {
          height: 5,
          width: 5
      },
  },

  notesBoxBody: {
    fontFamily:"Reenie Beanie",
    fontSize: 180
  }
});

export default Notes;