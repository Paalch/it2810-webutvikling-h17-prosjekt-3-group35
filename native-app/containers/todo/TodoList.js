import React from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import TodoElement from './TodoElement';

export default class TodoList extends React.Component {
    constructor() {
        super();
        this.state = {
            todos: [],
            text: ''
        };
        this.updateTextField = this.updateTextField.bind(this);
        this.saveTodo = this.saveTodo.bind(this);
        this.removeElement = this.removeElement.bind(this);
    }

    updateTextField(e) {
        this.setState({text: e.target.value});
    }

    removeElement(todo) {
        this.setState({
            todos: this.state.todos.filter((t) =>
                t !== todo
            )
        });
    }

    saveTodo() {
        const todoValue = this.state.text;
        if (todoValue.length === 0) {
            return;
        }
        this.setState({todos: this.state.todos.concat(todoValue), text: ''});
    };

    render() {
        const content = this.state.todos.map((todo, i) =>
            <TodoElement text={todo} key={i} removeElement={this.removeElement}/>
        );
        return (
            <View style={styles.container}>
                <View>
                    <View>
                        <TextInput
                            style={styles.formInput}
                            type={'text'}
                            placeholder={'.............................................................'}
                            value={this.state.text}
                            onChange={this.updateTextField}
                        />
                    </View>
                    <View>
                        <Button onPress={this.saveTodo} title={'+'}/>
                    </View>
                </View>
                <View style={styles.todos}>
                    {content}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#ccccff",
        padding: 20,
    },

    todos: {
        marginTop: 10,
    },

});
