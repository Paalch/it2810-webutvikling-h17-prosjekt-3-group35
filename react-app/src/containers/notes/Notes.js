import React from 'react';
import { Button } from 'react-bootstrap'
import './notes.css'
import NoteElement from "./NoteElement";

class Notes extends React.Component {
  constructor() {
    super();
    this.state = {
      notes: [],
      currentKeyCounter: 0
    }
    this.addNewNote = this.addNewNote.bind(this);
    this.deleteNote = this.deleteNote.bind(this);
    this.updateTextField = this.updateTextField.bind(this);
  }

  addNewNote() {
    this.setState({
      notes: this.state.notes.concat({id: this.state.currentKeyCounter, text: ''}),
      currentKeyCounter: this.state.currentKeyCounter + 1
    });
  }

  deleteNote(index) {
    const tempNotes = this.state.notes.filter(n => n.id !== index);
    this.setState({notes: tempNotes,currentKeyCounter: this.state.currentKeyCounter});
  }

  updateTextField(index, text) {
    const tempNotes = this.state.notes;
    tempNotes.forEach((note) => {
      if(note.id === index) {
        note.text = text;
      }
    });
    this.setState({notes: tempNotes, currentKeyCounter: this.state.currentKeyCounter})

  }

  render () {
    const content = this.state.notes.map((note, i) =>
      <NoteElement key={i} note={note} updateTextField={this.updateTextField} deleteNote={this.deleteNote}/>
    );

    return (
      <div>
        <Button bsStyle={'warning'} onClick={this.addNewNote}>+ new note</Button>
        <div>
          { content }
        </div>
      </div>
    )
  }
}

export default Notes;