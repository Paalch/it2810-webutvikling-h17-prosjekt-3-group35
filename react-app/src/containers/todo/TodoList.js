import React, {Component} from 'react';
import {Col, Row, FormControl, Button, FormGroup, InputGroup} from 'react-bootstrap';
import TodoElement from "./TodoElement";
import './todo.css';

class TodoList extends Component {

  constructor() {
    super();
    this.state = {
      todos: [],
      text: ''
    };
    this.updateTextField = this.updateTextField.bind(this);
    this.saveTodo = this.saveTodo.bind(this);
    this.removeElement = this.removeElement.bind(this);
    this.save = this.save.bind(this);
    this.load = this.load.bind(this);
  }

  componentDidMount() {
    const data = this.load();
    if (data !== null) {
      this.setState(() => data);
    }
  }

  componentDidUpdate() {
    this.save();
  }

  save() {
    localStorage.setItem('todo', JSON.stringify(this.state));
  }

  load() {
    return 'todo' in localStorage ?
      JSON.parse(localStorage.getItem('todo')) : null;
  }

  updateTextField(e) {
    this.setState({text: e.target.value});
  }

  removeElement(id) {
    this.setState({
      todos: this.state.todos.filter((t) =>
        id !== t.id
      )
    });
  }

  saveTodo() {
    const todoValue = this.state.text;
    if (todoValue.length === 0) {
      return;
    }
    const newTodo = {id: Date.now(), value: todoValue};
    this.setState({todos: this.state.todos.concat(newTodo), text: ''});
  };

  render() {
    const content = this.state.todos.map((element, i) =>
      <TodoElement todo={element} key={i} removeElement={this.removeElement}/>
    );

    return (
      <div>
        <Row>
          <Col xs={10}>
            <FormControl
              id={'todo-form-input'}
              type={'text'}
              placeholder={'.............................................................'}
              value={this.state.text}
              onChange={this.updateTextField}
            />
          </Col>
          <Col xs={2}>
            <Button
              className={'todo-add-button'}
              onClick={this.saveTodo}>
              +
            </Button>
          </Col>
        </Row>
        <div id={'todos'}>
          {content}
        </div>
      </div>
    )
  }
}

export default TodoList;