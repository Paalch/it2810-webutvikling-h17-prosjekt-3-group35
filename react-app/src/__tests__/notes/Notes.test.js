import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Notes from '../../containers/notes/Notes'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import TodoList from '../../containers/todo/TodoList'


Enzyme.configure({ adapter: new Adapter() })


describe('Checks the Note component', () => {

    //checks that todo render, bt compareing snapshot
    it('Checks that Todo renders', () => {
        const rendered = renderer.create(
            <Notes />
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });


    it('Checks that new notes can be added', () => {
        const wrapper = shallow(<Notes />)

        //Simulates 2 button clicks
        wrapper.find('Button').simulate('click');
        wrapper.find('Button').simulate('click');

        //Checks that 2 new notes are createsd
        let state = wrapper.state().notes
        expect(state).toHaveLength(2)
    })


});