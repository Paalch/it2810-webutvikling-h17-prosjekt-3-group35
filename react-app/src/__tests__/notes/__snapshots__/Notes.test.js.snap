// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks the Note component Checks that Todo renders 1`] = `
<div
  className="notes-container box"
>
  <div
    className="notes-header box-header row"
  >
    <div
      className="col-xs-7"
      style={
        Object {
          "paddingLeft": "0px",
        }
      }
    >
      <h1>
        Notes
      </h1>
    </div>
    <div
      className="col-xs-5"
    >
      <button
        className="new-note-button btn btn-warning"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        + New note
      </button>
    </div>
  </div>
  <div
    className="notes-body box-body row"
  >
    <div />
  </div>
</div>
`;
